app/src/main/java/kth/jjve/xfran/EditProfileActivity.java

```diff
@@ -0,0 +1,138 @@
+package kth.jjve.xfran;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.ActionBarDrawerToggle;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.appcompat.widget.Toolbar;
+import androidx.core.view.GravityCompat;
+import androidx.drawerlayout.widget.DrawerLayout;
+import androidx.lifecycle.ViewModelProviders;
+
+import android.annotation.SuppressLint;
+import android.content.Intent;
+import android.media.Image;
+import android.os.Bundle;
+import android.view.MenuItem;
+import android.widget.EditText;
+import android.widget.ImageButton;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.google.android.material.navigation.NavigationView;
+
+import kth.jjve.xfran.viewmodels.UserProfileVM;
+
+public class EditProfileActivity extends AppCompatActivity implements NavigationView.OnNavigationItemSelectedListener {
+    private EditText mFirstName, mLastName, mWeight, mLength;
+    private TextView mEmail;
+    private DrawerLayout drawerLayout;
+    private NavigationView navigationView;
+    private Toolbar toolbar;
+
+    @SuppressLint("SetTextI18n")
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_edit_profile);
+
+        /*----- HOOKS -----*/
+        drawerLayout = findViewById(R.id.drawer_layout_editprofile);
+        navigationView = findViewById(R.id.nav_view_editprofile);
+        toolbar = findViewById(R.id.editprofile_toolbar);
+
+        mFirstName = findViewById(R.id.editProfileFirstName);
+        mLastName = findViewById(R.id.editProfileLastName);
+        mWeight = findViewById(R.id.editProfileWeight);
+        mLength = findViewById(R.id.editProfileLength);
+        mEmail = findViewById(R.id.editProfileEmail);
+        ImageButton save = findViewById(R.id.saveProfile);
+
+        /*-----  VM  -----*/
+        //Todo: check if we can somehow inherit this between editprofile and normal profile
+        UserProfileVM mUserProfileVM = ViewModelProviders.of(this).get(UserProfileVM.class);
+        mUserProfileVM.init();
+        mUserProfileVM.getUserProfile().observe(this, uP -> {
+            if (!uP.checkEmpty()){
+                mFirstName.setText(uP.getFirstName());
+                mLastName.setText(uP.getLastName());
+                mEmail.setText(uP.getEmail());
+                mWeight.setText(Double.toString(uP.getWeight()));
+                mLength.setText(Integer.toString(uP.getLength()));
+            }
+        });
```

Comment:
Could this be made into a method? I mean it wouldn't change anything, except making the onCreate a bit shorter, which I find less confusing

Reply:
I think line 52-54 should be in onCreate but line 55-60 could be a method; yes